app/src/main/java/tech/relaycorp/letro/conversation/server/processor/NewMessageProcessor.kt

```diff
@@ -80,4 +80,16 @@ class NewMessageProcessorImpl @Inject constructor(
             ),
         )
     }
+
+    override suspend fun isFromExpectedSender(
+        message: IncomingMessage,
+        awalaManager: AwalaManager,
+    ): Boolean {
+        val messageWrapper = (parser.parse(message.content) as NewMessageIncomingMessage).content
```

Comment:
It can be problematic to parse the same `IncomingMessage` twice -- once to run `isFromExpectedSender()` and then again to actually process it. Messages like VeraId bundles and conversations/replies with many attachments will be particularly expensive, so we shouldn't compound this by parsing them twice.

I think we should attempt to parse any `IncomingMessage` whose `type` we recognise, and then pass the result to `isFromExpectedSender()` along with the `IncomingMessage` and `AwalaManager`. For example:

```kotlin
interface NewMessageProcessor : AwalaMessageProcessor<LetroReply>

class NewMessageProcessorImpl @Inject constructor(
    private val pushManager: PushManager,
    private val parser: NewMessageMessageParser,
    // ...,
) : NewMessageProcessor {
    // ...
    override suspend fun isFromExpectedSender(
        content: LetroReply,
        message: IncomingMessage,
        awalaManager: AwalaManager,
    ): Boolean {
        // ...
    }
    // ...
}
```

Thoughts?